packages/core/src/test/codewhisperer/commands/startSecurityScan.test.ts

```diff
@@ -449,3 +449,104 @@ describe('startSecurityScan', function () {
+        sinon.stub(timeoutUtils, 'sleep')
+    })
+    afterEach(function () {
+        sinon.restore()
+    })
+    after(async function () {
+        await closeAllEditors()
+    })
+    const createClient = () => {
+        const mockClient = stub(DefaultCodeWhispererClient)
+
+        mockClient.createCodeScan.resolves(mockCreateCodeScanResponse)
+        mockClient.createUploadUrl.resolves(mockCreateUploadUrlResponse)
+        mockClient.getCodeScan.resolves(mockGetCodeScanResponse)
+        mockClient.listCodeScanFindings.resolves(mockListCodeScanFindingsResponse)
+        return mockClient
+    }
+
+    const openTestFile = async (filePath: string) => {
+        const doc = await vscode.workspace.openTextDocument(filePath)
+        return await vscode.window.showTextDocument(doc, {
+            selection: new vscode.Range(new vscode.Position(0, 0), new vscode.Position(0, 1)),
+        })
+    }
+
+    it('Should calculate cpu and memory usage for file scans', async function () {
+        getFetchStubWithResponse({ status: 200, statusText: 'testing stub' })
+        const commandSpy = sinon.spy(vscode.commands, 'executeCommand')
+        const securityScanRenderSpy = sinon.spy(diagnosticsProvider, 'initSecurityScanRender')
+
+        await model.CodeScansState.instance.setScansEnabled(true)
+        const startScanCpuUsage = process.cpuUsage()
+
+        await startSecurityScan.startSecurityScan(
+            mockSecurityPanelViewProvider,
+            editor,
+            createClient(),
+            extensionContext,
+            CodeAnalysisScope.FILE
+        )
+
+        // EndScanCpuUsageByUser is the difference of CPU Usage from start of a scan to end of a scan.
+        const EndScanCpuUsage = process.cpuUsage(startScanCpuUsage)
+        const EndScanCpuUsageByUser = EndScanCpuUsage.user / 1000000 // Convert microseconds to seconds
+
+        /**
+         * CPU Usage at start of a scan
+         * const startScanCpuUsageByUser = startScanCpuUsage.user / 1000000
+         * We can calculate system CPU usage but this will vary depends of ongoing background activity so its irrelevant.
+         *  const startScanCpuUsageBySystem = startScanCpuUsage.system / 1000000
+         *  const EndScanCpuUsageBySystem = EndScanCpuUsage.system / 1000000
+         */
+
+        const EndScanMemoryUsage = process.memoryUsage().heapTotal
+        const EndScanMemoryUsageInMB = EndScanMemoryUsage / (1024 * 1024) // Converting bytes to MB
+
+        /**
+            Input:-
+            Payload Size is 198.99
+            Result:-
+            Scan Memory Usage at end: 203.75390625 MB
+            Start of a Scan:
+                User CPU Usage: 21.446874 sec and System CPU Usage: 7.044477 sec
+            Difference at End of a Scan:
+                User CPU Usage: 0.058255 sec and System CPU Usage: 0.004867 sec
+         */
+
+        assert(
+            EndScanCpuUsageByUser > 0,
```

Comment:
Maybe I don't fully understand what is being captured here but this can never be 0 or less right? Since you can't have "negative" CPU usage.
Shouldn't this be converting a percentage of CPU usage and validating it against some baseline, e.g. cpuUsage < 50%

Reply:
calculated and tested CPU usage in 6 different machines and made note in the summary.